src/Corax/IndexWriter.cs

```diff
@@ -217,15 +282,51 @@ private void DeleteCommit(Dictionary<Slice, int> knownFields, Span<byte> tmpBuf,
                     if (fieldType.HasFlag(IndexEntryFieldType.List))
                     {
                         var it = entryReader.ReadMany(fieldId);
+                        
                         while (it.ReadNext())
                         {
-                            DeleteField(id, fieldName, tmpBuf, it.Sequence);
+                            if (analyzer == null)
+                            {
+                                DeleteField(id, fieldName, tmpBuf, it.Sequence);
+                            }
+                            else
+                            {
+                                // Because of how we store the data, either this is a sequence or a tuple, which also contains a sequence. 
+                                var value = it.Sequence;
+
+                                var words = new Span<byte>(tempWordsSpace, bufferSize);
+                                var tokens = new Span<Token>(tempTokenSpace, tokenSize);
```

Comment:
this is duplicated from the `Index`, can we avoid it?

Reply:
@redknightlois ?

Reply:
No, they may change over the `Execute(_, ref words, ref tokens )` call. We create them every time by design.